clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancer.groovy

```diff
@@ -22,8 +22,33 @@ import com.netflix.spinnaker.clouddriver.azure.AzureCloudProvider
 import com.netflix.spinnaker.clouddriver.model.LoadBalancer
 import com.netflix.spinnaker.clouddriver.model.LoadBalancerServerGroup
 import com.netflix.spinnaker.moniker.Moniker
+import groovy.transform.CompileStatic
 
+@CompileStatic
 class AzureLoadBalancer implements LoadBalancer {
+  static enum AzureLoadBalancerType {
+    AZURE_LOAD_BALANCER,
+    AZURE_APPLICATION_GATEWAY
+
+    @Override
+    String toString() {
```

Comment:
This seems like a fairly complicated routine for returning what are essentially constants. Why not just define an additional field on the enum and populate it on construction of each enum value?

Reply:
Something like this (Java here for the example):

```
  public enum AzureLoadBalancerType {
    AZURE_LOAD_BALANCER("Azure Load Balancer"),
    AZURE_APPLICATION_GATEWAY("Azure Application Gateway");

    private String formattedType;

    private AzureLoadBalancerType(String formattedType) {
      this.formattedType = formattedType;
    }

    public String toString() {
      return formattedType;
    }
  }
```

Reply:
That looks like more concise